pkg/cmd/cli/cmd/logs_test.go

```diff
@@ -30,3 +37,163 @@ func TestLogsFlagParity(t *testing.T) {
 		}
 	})
 }
+
+type fakeBuildClient struct {
+	build *buildapi.Build
+}
+
+func (f *fakeBuildClient) List(opts metav1.ListOptions) (*buildapi.BuildList, error) {
+	return nil, nil
+}
+
+func (f *fakeBuildClient) Get(names string, opts metav1.GetOptions) (*buildapi.Build, error) {
+	return f.build, nil
+}
+
+func (f *fakeBuildClient) Create(build *buildapi.Build) (*buildapi.Build, error) {
+	return nil, nil
+}
+
+func (f *fakeBuildClient) Update(build *buildapi.Build) (*buildapi.Build, error) {
+	return nil, nil
+}
+
+func (f *fakeBuildClient) Delete(name string) error {
+	return nil
+}
+
+func (f *fakeBuildClient) Watch(opts metav1.ListOptions) (watch.Interface, error) {
+	return nil, nil
+}
+
+func (f *fakeBuildClient) Clone(request *buildapi.BuildRequest) (*buildapi.Build, error) {
+	return nil, nil
+}
+
+func (f *fakeBuildClient) UpdateDetails(build *buildapi.Build) (*buildapi.Build, error) {
+	return nil, nil
+}
+
+type fakeNamespacer struct {
+	client buildclient.BuildInterface
+}
+
+func (f *fakeNamespacer) Builds(namespace string) buildclient.BuildInterface {
+	return f.client
+}
+
+type fakeWriter struct {
+	data []byte
+}
+
+func (f *fakeWriter) Write(p []byte) (n int, err error) {
+	f.data = p
+	return len(p), nil
+}
+
+func TestRunLogForPipelineStrategy(t *testing.T) {
+	bld := buildapi.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "foo",
+			Annotations: map[string]string{buildapi.BuildJenkinsBlueOceanLogURLAnnotation: "https://foo"},
+		},
+		Spec: buildapi.BuildSpec{
+			CommonSpec: buildapi.CommonSpec{
+				Strategy: buildapi.BuildStrategy{
+					JenkinsPipelineStrategy: &buildapi.JenkinsPipelineBuildStrategy{},
+				},
+			},
+		},
+	}
+
+	fakebc := fakeBuildClient{
+		build: &bld,
+	}
+	fakenamespacer := fakeNamespacer{
+		client: &fakebc,
+	}
+	fakewriter := fakeWriter{}
+
+	opts := OpenShiftLogsOptions{
+		KubeLogOptions: &kcmd.LogsOptions{
+			Object: &bld,
```

Comment:
yeah meant to do that and go side tracked ... the part with the fake build getter is there